Code/GraphMol/MolStandardize/Tautomer.cpp

```diff
@@ -75,35 +72,13 @@ int scoreRings(const ROMol &mol) {
   return score;
 };
 
-struct smarts_mol_holder {
-  std::string d_smarts;
-  ROMOL_SPTR dp_mol;
-  smarts_mol_holder(const std::string &smarts) : d_smarts(smarts) {
-    dp_mol.reset(SmartsToMol(smarts));
-  }
-};
-
-typedef boost::flyweight<
-    boost::flyweights::key_value<std::string, smarts_mol_holder>,
-    boost::flyweights::no_tracking>
-    smarts_mol_flyweight;
-
-struct SubstructTerm {
-  std::string name;
-  std::string smarts;
-  int score;
-  ROMOL_SPTR matcher;
-  SubstructTerm(std::string aname, std::string asmarts, int ascore)
-      : name(std::move(aname)), smarts(std::move(asmarts)), score(ascore) {
-    matcher = smarts_mol_flyweight(smarts).get().dp_mol;
-  };
-};
+SubstructTerm::SubstructTerm(std::string aname, std::string asmarts, int ascore)
+  : name(std::move(aname)), smarts(std::move(asmarts)), score(ascore), matcher(SmartsToMol(smarts)) {
+}
 
-int scoreSubstructs(const ROMol &mol) {
-  // a note on efficiency here: we'll construct the SubstructTerm objects here
-  // repeatedly, but the SMARTS parsing for each entry will only be done once
-  // since we're using the boost::flyweights above to cache them
-  const std::vector<SubstructTerm> substructureTerms{
+const std::vector<SubstructTerm> &getDefaultTautomerSubstructs()
```

Comment:
I think `getDefaultTautomerScoreSubstructs()` would be a clearer name for this

Reply:
Done